app/models/pipeline_run_stage.rb

```diff
@@ -26,6 +26,8 @@ class PipelineRunStage < ApplicationRecord
   # Max number of times we resubmit a job when it gets killed by EC2.
   MAX_RETRIES = 5
 
+  after_create :create_dag
```

Comment:
Our system currently abuses using events, which I consider very harmful, because everything has a "side effect".

For example, I noticed method `create_dag` has a `save` method, which means the record will be saved twice in a row. Not exactly harmful in this specific situation, but just to show what kind of side effects we are talking about. For example: if someone decides adding a `after_save` later on, this person will have a hard time to trying to understand why it is being triggered twice.

I would rather make `create_dag` method to be stateless (i.e. receiving all parameters that it needs to create), and probably rename it to something like `generate_dag_json` (or even better, extract it to a helper class), and then invoke it explicitly in this block:
https://github.com/chanzuckerberg/idseq-web/blob/ca0b34ab1e68ae8c1d3455860c0b2408e531148e/app/models/pipeline_run.rb#L317-L349

Reply:
Updated with `dag_json_helper`